persistence/src/main/scala/za/co/absa/spline/persistence/ArangoDatabaseFacade.scala

```diff
@@ -51,7 +51,9 @@ class ArangoDatabaseFacade(connectionURL: ArangoConnectionURL) extends Disposabl
     arangoBuilder.build
   }
 
-  val db: ArangoDatabaseAsync = {
+  // The val should be lazy to not prevent a facade instance from being created.
+  // It would allow connection to be re-attempted later and the {{shutdown()}} method to be called.
+  lazy val db: ArangoDatabaseAsync = {
```

Comment:
I would simplify it: `The val is lazy ...` and `It allows connection ...`

Reply:
done